src/v0/sources/shopify_v2/identityResolutionLayer.js

```diff
@@ -0,0 +1,87 @@
+const { v5 } = require('uuid');
+const { SHOPIFY_ADMIN_ONLY_EVENTS } = require('./config');
+const { getCartToken } = require('./commonUtils');
+const { generateUUID, isDefinedAndNotNull } = require('../../util');
+
+const idResolutionLayer = {
+  /**
+   * This function checks and returns rudderId from message if present
+   * returns null if not present or found
+   * @param {*} message
+   */
+  getRudderIdFromNoteAtrributes(noteAttributes, field) {
+    const rudderIdObj = noteAttributes.find((obj) => obj.name === field);
+    if (isDefinedAndNotNull(rudderIdObj)) {
+      return rudderIdObj.value;
+    }
+    return null;
+  },
+
+  /**
+   * This function retrieves anonymousId and sessionId in folowing steps:
+   * 1. Checks for `rudderAnonymousId`and `rudderSessionId in `note_atrributes`
+   * 2. Checks in dbData
+   * 3. This means we don't have `anonymousId` and hence events CAN NOT be stitched and we check for cartToken
+   *    a. if cartToken is available we return its hash value
+   *    b. else we check if the event is an SHOPIFY_ADMIN_ONLY_EVENT
+   *       -> if true we return `null`;
+   *       -> else we don't have any identifer (very edge case) we return `random anonymousId`
+   *    No Random SessionId is generated as its not a required field
+   * @param {*} message
+   * @param {*} metricMetadata
+   * @returns
+   */
+  getAnonymousIdAndSessionId(message, shopifyTopic, dbData = null) {
+    let anonymousId;
+    let sessionId;
+    const noteAttributes = message.properties?.note_attributes;
+    // Giving Priority to note_attributes to fetch rudderAnonymousId over Redis due to better efficiency
+    if (isDefinedAndNotNull(noteAttributes)) {
+      anonymousId = this.getRudderIdFromNoteAtrributes(noteAttributes, 'rudderAnonymousId');
+      sessionId = this.getRudderIdFromNoteAtrributes(noteAttributes, 'rudderSessionId');
+      if (anonymousId && sessionId) {
+        return { anonymousId, sessionId };
+      }
+    }
+
+    // falling back to cartToken mapping or its hash in case no rudderAnonymousId or rudderSessionId is found
+    const cartToken = getCartToken(message, shopifyTopic);
+    if (!isDefinedAndNotNull(cartToken)) {
```

Comment:
Can't we validate using dbData since we already have that? Why are we validating with cartToken here?